projects/storefrontlib/router/on-navigate.service.ts

```diff
@@ -51,6 +51,8 @@ export class OnNavigateService {
     this.subscription?.unsubscribe();
 
     if (enable) {
+      // Disable automatic scroll restoration to avoid race conditions
+      this.viewportScroller.setHistoryScrollRestoration('manual');
```

Comment:
Even though this change is small and works well, we need to add feature flags for `5.1` and mention it in change logs as it is still breaking. As well as leaving cleanup comments to remove the logic for `6.0`.

Reply:
@Zeyber 
added feature flag .. thanks